### ISSUE TITLE
Child replication issue when ReplicationTarget is changed on parent

### ISSUE THREAD

**Issue by Skippeh** [cBournhonesque/lightyear#649]
Child entities with the `Replicate` component are not replicated along with parent when a new client is added to the parent's `ReplicationTarget` component

**Comment by cBournhonesque** [comment]
Hm normally it should work.
Is this your situation?
1) spawning P1 with child C1, both have `Replicate`
2) you update the `ReplicationTarget` on P1 to add a new client
3) The `ReplicationTarget` is updated on C1 but C1 is still not replicated to the new client?

**Comment by Skippeh** [comment]
I have P1 with `ReplicationTarget { target: NetworkTarget::Only(...) }` and C1 with NetworkTarget::All

When i add more clients to P1, C1 is unchanged but should still replicate to the new clients that are added to P1, right?

**Comment by cBournhonesque** [comment]
And you have `replicate_hierarchy: True`?
I feel like if you add more clients to P1, then C1 will get the same `ReplicationTarget` and P1 and will also get replicated to the new client; but maybe there's a bug, it would require more testing

**Comment by Skippeh** [comment]
Yes, `replicate_hierarchy` is set to true. It works fine if the client is added to P1's network targets when the component is added. It only fails when P1's network targets is updated.

Also,
> and P1 and will also get replicated to the new client

P1 *is* replicated to the new client, but C1 is not.

**Comment by cBournhonesque** [comment]
You're correct that the change to the ReplicationTarget of P1 isn't propagated to C1!
I have a fix for it, but you won't be abel to have a different ReplicationTarget in P1 and C1, any change in P1 (for example changing the ReplicationTarget) will be propagated to all children.
